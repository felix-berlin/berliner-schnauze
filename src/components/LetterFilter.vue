<template>
  <nav class="c-letter-filter">
    <ul class="c-letter-filter__list u-list-reset">
      <!-- <li
        class="c-letter-filter__list-item"
        :class="{ 'is-current': $store.state.wordFilteredByLetter === null }"
      >
        <button
          class="c-letter-filter__button c-letter-filter__button--all c-button"
          type="button"
          @click="$store.dispatch('filterByLetter', null)"
        >
          Alle
        </button>
      </li> -->
      <li
        v-for="(letter, index) in props.groups"
        :key="index"
        class="c-letter-filter__list-item"
        :class="{ 'is-current': letter === wordFilteredByLetter }"
      >
        <button
          :aria-label="'Filter nach Buchstabe ' + letter"
          type="button"
          class="c-letter-filter__button c-button"
          :title="'Filter nach Buchstabe ' + letter"
          @click="updateLetterFilter(letter)"
        >
          {{ letter }}
        </button>
      </li>
    </ul>
  </nav>
</template>

<script setup lang="ts">
import { computed, ref } from "vue";

interface LetterFilterProps {
  groups: string[];
}

const props = defineProps<LetterFilterProps>();

const wordFilteredByLetter = ref("");

const updateLetterFilter = (letter: string): void => {
  console.log("letter", letter);
};
</script>

<style lang="scss">
@use "@styles/components/letter-filter";
</style>
