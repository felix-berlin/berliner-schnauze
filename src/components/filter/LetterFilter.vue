<template>
  <nav class="c-letter-filter">
    <ul class="c-letter-filter__list u-list-reset">
      <li class="c-letter-filter__list-item">
        <button
          class="c-letter-filter__button is-all c-button c-button--filter"
          :class="{ 'is-current': wordSearchStore.activeLetterFilter === '' }"
          type="button"
          @click="setLetterFilter('')"
        >
          Alle
        </button>
      </li>
      <li
        v-for="letter in wordSearchStore.letterGroups"
        :key="letter!"
        class="c-letter-filter__list-item"
      >
        <button
          :aria-label="`Filter nach Buchstabe ${letter}`"
          type="button"
          class="c-letter-filter__button c-button c-button--filter"
          :class="{ 'is-current': letter === wordSearchStore.activeLetterFilter }"
          :title="`Filter nach Buchstabe ${letter}`"
          @click="setLetterFilter(letter!)"
        >
          {{ letter }}
        </button>
      </li>
    </ul>
  </nav>
</template>

<script setup lang="ts">
import { computed, ref } from "vue";
import { useStore } from "@nanostores/vue";
import { $wordSearch, setLetterFilter } from "@stores/index.ts";

const wordSearchStore = useStore($wordSearch);
</script>

<style lang="scss">
@use "@styles/components/letter-filter";
</style>
