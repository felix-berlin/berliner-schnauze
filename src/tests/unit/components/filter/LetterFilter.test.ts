import { describe, it, expect, vi, beforeEach } from "vitest";
import { mount, VueWrapper } from "@vue/test-utils";
import LetterFilter from "@components/filter/LetterFilter.vue";
import { nextTick } from "vue";
import { $wordSearch, setLetterFilter } from "@stores/index.ts";
import { useStore } from "@nanostores/vue";

vi.mock("@nanostores/vue", () => ({
  useStore: () => ({
    activeLetterFilter: "",
    letterGroups: ["a", "b", "c"],
  }),
}));

vi.mock("@stores/index.ts", () => ({
  $wordSearch: vi.fn(),
  setLetterFilter: vi.fn(),
}));

describe("LetterFilter.vue", () => {
  let wrapper: VueWrapper<any, any>;

  beforeEach(() => {
    wrapper = mount(LetterFilter);
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("calls setLetterFilter when a button is clicked", async () => {
    const button = wrapper.find(".c-letter-filter__button");
    await button.trigger("click");
    expect(setLetterFilter).toHaveBeenCalled();
  });
});
