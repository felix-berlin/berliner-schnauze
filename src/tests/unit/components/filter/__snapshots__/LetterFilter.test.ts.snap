// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LetterFilter.vue > renders correctly 1`] = `
<nav
  class="c-letter-filter"
>
  <ul
    class="c-letter-filter__list u-list-reset"
  >
    <li
      class="c-letter-filter__list-item"
    >
      <button
        class="c-letter-filter__button is-all c-button c-button--filter is-current"
        type="button"
      >
         Alle 
      </button>
    </li>
    
    <li
      class="c-letter-filter__list-item"
    >
      <button
        aria-label="Filter nach Buchstabe a"
        class="c-letter-filter__button c-button c-button--filter"
        title="Filter nach Buchstabe a"
        type="button"
      >
        a
      </button>
    </li>
    <li
      class="c-letter-filter__list-item"
    >
      <button
        aria-label="Filter nach Buchstabe b"
        class="c-letter-filter__button c-button c-button--filter"
        title="Filter nach Buchstabe b"
        type="button"
      >
        b
      </button>
    </li>
    <li
      class="c-letter-filter__list-item"
    >
      <button
        aria-label="Filter nach Buchstabe c"
        class="c-letter-filter__button c-button c-button--filter"
        title="Filter nach Buchstabe c"
        type="button"
      >
        c
      </button>
    </li>
    
  </ul>
</nav>
`;
